src/caip-types.test-d.ts

```diff
@@ -0,0 +1,52 @@
+import { expectAssignable, expectNotAssignable } from 'tsd';
+
+import type {
+  CaipAccountAddress,
+  CaipAccountId,
+  CaipChainId,
+  CaipNamespace,
+  CaipReference,
+} from '.';
+
+const embeddedString = 'test';
+
+// Valid caip strings:
+
+expectAssignable<CaipChainId>('namespace:reference');
+expectAssignable<CaipChainId>('namespace:');
```

Comment:
I'm assuming that this is non-enforceable via TypeScript, is that correct?

Reply:
Unfortunately not. We could try to hack it by doing something like this

```
type Character = "a" | "b" | "c" | "d" | "e" | "f" | "g" | ...

export type CaipChainId = `${Character}${string}:${Character}${string}`;
```

But I believe this makes a type with n^2 definitions since [they get crossmulitplied](https://www.typescriptlang.org/docs/handbook/2/template-literal-types.html)